packages/app/src/NodeDetailsSidePanel.tsx

```diff
@@ -0,0 +1,286 @@
+    isUTC: false,
+    defaultValue: 'Past 1h',
+    defaultTimeRange: [
+      defaultTimeRange?.[0]?.getTime() ?? -1,
+      defaultTimeRange?.[1]?.getTime() ?? -1,
+    ],
+  });
+
+  const handleClose = React.useCallback(() => {
+    setNodeName(undefined);
+  }, [setNodeName]);
+
+  if (!nodeName) {
+    return null;
+  }
+
+  return (
+    <Drawer
+      enableOverlay
+      overlayOpacity={0.1}
+      duration={0}
+      open={!!nodeName}
+      onClose={handleClose}
+      direction="right"
+      size={'80vw'}
+      zIndex={drawerZIndex}
+    >
+      <ZIndexContext.Provider value={drawerZIndex}>
+        <div className={styles.panel}>
+          <DrawerHeader
+            header={`Details for ${nodeName}`}
+            onClose={handleClose}
+          />
+          <DrawerBody>
+            <Grid>
+              <NodeDetails name={nodeName} dateRange={dateRange} />
+              <Grid.Col span={6}>
+                <Card p="md">
+                  <Card.Section p="md" py="xs" withBorder>
+                    CPU Usage by Pod
+                  </Card.Section>
+                  <Card.Section p="md" py="sm" h={CHART_HEIGHT}>
+                    <HDXLineChart
+                      config={{
+                        dateRange,
+                        granularity: convertDateRangeToGranularityString(
+                          dateRange,
+                          60,
+                        ),
+                        groupBy: 'k8s.pod.name',
+                        where,
+                        table: 'metrics',
+                        aggFn: 'avg',
+                        field: 'k8s.pod.cpu.utilization - Gauge',
+                        numberFormat: K8S_CPU_PERCENTAGE_NUMBER_FORMAT,
+                      }}
+                    />
+                  </Card.Section>
+                </Card>
+              </Grid.Col>
+              <Grid.Col span={6}>
+                <Card p="md">
+                  <Card.Section p="md" py="xs" withBorder>
+                    Memory Usage by Pod
+                  </Card.Section>
+                  <Card.Section p="md" py="sm" h={CHART_HEIGHT}>
+                    <HDXLineChart
+                      config={{
+                        dateRange,
+                        granularity: convertDateRangeToGranularityString(
+                          dateRange,
+                          60,
+                        ),
+                        groupBy: 'k8s.pod.name',
+                        where,
+                        table: 'metrics',
+                        aggFn: 'avg',
+                        field: 'k8s.pod.memory.usage - Gauge',
+                        numberFormat: K8S_MEM_NUMBER_FORMAT,
+                      }}
+                    />
+                  </Card.Section>
+                </Card>
+              </Grid.Col>
+              <Grid.Col span={12}>
+                <InfraPodsStatusTable dateRange={dateRange} where={where} />
```

Comment:
When clicking on the pod table here, the side panel gets obscured behind since it's triggering it in the "wrong" z index context. I wonder if we still need to import side panels like we do with the log side panel (along with the table, instead of just globally on a page).

Reply:
i added a hack to check if the panel is nested, but going to rework it with a separate state later